File: financial-manager-api/Entities/Requests/TransactionRequest.cs
using System.ComponentModel.DataAnnotations;

namespace financial_manager.Entities.Requests
{
    public class TransactionRequest
    {
        public int? Id { get; set; }
        [Required]
        [MinLength(5)]
        public string Title { get; set; } = null!;
        [Required]
        [Range(1, 99999.99)]
        public decimal Significance { get; set; }
        [Required]
        public string TransactionType { get; set; } = null!;
        public DateTime? CreatedAt { get; set; } = DateTime.Now;
        public DateTime? ExpenseDate { get; set; } = DateTime.Now;
        public CategoryRequest? Category { get; set; }
    }
}

File: financialManagerUI/src/app/entities/requests/transaction.request.ts
import { BaseEntity } from '../shared/baseEntity';
import { CategoryRequest } from './category.request';

export class TransactionRequest extends BaseEntity {
  public title: string;
  public significance: number;
  public transactionType: string;
  public createdAt?: Date;
  public expenseDate?: Date;
  public category?: CategoryRequest = undefined;

  constructor(
    title: string,
    significance: number,
    transactionType: string,
    expenseDate?: Date,
    category?: CategoryRequest,
    id?: number
  ) {
    super();
    this.title = title;
    this.significance = significance;
    this.transactionType = transactionType;
    this.expenseDate = expenseDate;
    this.category = category;
    this.id = id;
  }
}
